### PR TITLE
add openexr 2.3.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **openexr/2.3.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
https://github.com/conan-io/conan-center-index/pull/204

File: recipes/openexr/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+class OpenEXRConan(ConanFile):
+    name = "openexr"
+    description = "OpenEXR is a high dynamic-range (HDR) image file format developed by Industrial Light & " \
+                  "Magic for use in computer imaging applications."
+    license = "BSD"
+    homepage = "https://github.com/openexr/openexr"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "namespace_versioning": [True, False], "fPIC": [True, False]}
+    default_options = "shared=False", "namespace_versioning=True", "fPIC=True"
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
    default_options = {"shared": False, "namespace_versioning": True, "fPIC": True}
```

File: recipes/openexr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+from conans import ConanFile, CMake, tools
+import os
+import glob
+
+
+class OpenEXRConan(ConanFile):
+    name = "openexr"
+    description = "OpenEXR is a high dynamic-range (HDR) image file format developed by Industrial Light & " \
+                  "Magic for use in computer imaging applications."
+    license = "BSD"
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
    license = "BSD-3-Clause"
```

File: recipes/openexr/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def requirements(self):
+        self.requires('zlib/1.2.11')
+
+    def configure(self):
+        if "fPIC" in self.options.fields and self.options.shared:
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]: I'm not sure if you should force. I know that doesn't make sense building shared + non-fPIC, but it's a generic rule. If we accept this, we should update ALL recipes.

File: recipes/openexr/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            self.options.fPIC = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        tools.replace_in_file(os.path.join('openexr-{}'.format(self.version), 'CMakeLists.txt'),
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]: maybe we could create a patch file for it. It will look neat.

File: recipes/openexr/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+set(OPENEXR_PACKAGE_PREFIX ${CMAKE_INSTALL_PREFIX})
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+""")
+        for p in self.conan_data["patches"][self.version]:
+            tools.patch(**p)
+
+    def build(self):
+        # parallel builds are disabled due to the random issue: 'toFloat.h': No such file or directory
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]: nice comment!

File: recipes/openexr/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        cmake = CMake(self, parallel=self.settings.os != 'Windows')
+        cmake.definitions["OPENEXR_BUILD_PYTHON_LIBS"] = False
+        cmake.definitions["OPENEXR_BUILD_SHARED"] = self.options.shared
+        cmake.definitions["OPENEXR_BUILD_STATIC"] = not bool(self.options.shared)
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
        cmake.definitions["OPENEXR_BUILD_STATIC"] = not self.options.shared
```
Don't need to convert again.

File: recipes/openexr/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                              'IlmImfUtil' + version_suffix,
+                              'IlmThread' + version_suffix,
+                              'Iex' + version_suffix,
+                              'Half' + version_suffix]
+        
+        self.cpp_info.includedirs = [os.path.join('include', 'OpenEXR'), 'include']
+        if self.options.shared and self.settings.os == "Windows":
+            self.cpp_info.defines.append("OPENEXR_DLL")
+
+        if self.settings.os != "Windows":
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
        if self.settings.os == "Linux":
```

File: recipes/openexr/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+            self.cpp_info.cppflags = ["-pthread"]
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
            self.cpp_info.libs.append("pthread")
```

File: recipes/openexr/all/test_package/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,13 @@
+project(PackageTest)
+cmake_minimum_required(VERSION 2.8)
+
+set(CMAKE_CXX_STANDARD 11)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+message(${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/openexr/all/test_package/CMakeLists.txt]:
```suggestion
```

File: recipes/openexr/all/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,13 @@
+project(PackageTest)
+cmake_minimum_required(VERSION 2.8)
+
+set(CMAKE_CXX_STANDARD 11)
```
Comment (uilianries) [inline recipes/openexr/all/test_package/CMakeLists.txt]:
```suggestion
```

File: recipes/openexr/all/test_package/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+message(${CONAN_LIBS})
+
+add_executable(testPackage main.cpp)
+target_link_libraries(testPackage ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/openexr/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(testPackage ${CONAN_LIBS})
set_property(TARGET testPackage PROPERTY CXX_STANDARD 11)
```

File: recipes/openexr/all/test_package/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,13 @@
+project(PackageTest)
+cmake_minimum_required(VERSION 2.8)
```
Comment (uilianries) [inline recipes/openexr/all/test_package/CMakeLists.txt]:
```suggestion
project(PackageTest)
```

File: recipes/openexr/all/test_package/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,13 @@
+project(PackageTest)
```
Comment (uilianries) [inline recipes/openexr/all/test_package/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 2.8)
```

File: recipes/openexr/all/test_package/conanfile.py [hunk 14]
```diff
@@ -0,0 +1,22 @@
+from conans.model.conan_file import ConanFile
```
Comment (uilianries) [inline recipes/openexr/all/test_package/conanfile.py]:
```suggestion
```

File: recipes/openexr/all/test_package/conanfile.py [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+from conans import CMake, tools
```
Comment (uilianries) [inline recipes/openexr/all/test_package/conanfile.py]:
```suggestion
from conans import ConanFile, CMake, tools
```

File: recipes/openexr/all/test_package/conanfile.py [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+class DefaultNameConan(ConanFile):
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def imports(self):
```
Comment (uilianries) [inline recipes/openexr/all/test_package/conanfile.py]: Remove imports, `run_environment=True` should be enough for finding dynamic libraries.

File: recipes/openexr/all/CMakeLists.txt.patch [hunk 17]
```diff
@@ -0,0 +1,18 @@
+--- CMakeLists.txt.orig 2018-08-10 10:34:26.000000000 +0200
++++ CMakeLists.txt
+@@ -13,6 +13,14 @@
+ set(OPENEXR_SOVERSION 24)
+
+ project(OpenEXR VERSION ${OPENEXR_VERSION})
++include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
++conan_basic_setup(KEEP_RPATHS)
```
Comment (danimtb) [inline recipes/openexr/all/CMakeLists.txt.patch]: Why is this KEEP_RPATHS needed?

Comment (danimtb) [reply recipes/openexr/all/CMakeLists.txt.patch]: @ericLemanissier ping

Comment (danimtb) [reply recipes/openexr/all/CMakeLists.txt.patch]: I don't think it is necessary anymore. Could you please give it a go removing it? I think it worths a try. Thanks!

**Comment by danimtb** [comment]
ok, the output is clear. Let's keep that patch @ericLemanissier and thanks a lot for giving it a try 😄

File: recipes/openexr/all/conanfile.py [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+    description = "OpenEXR is a high dynamic-range (HDR) image file format developed by Industrial Light & " \
+                  "Magic for use in computer imaging applications."
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/openexr/openexr"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "namespace_versioning": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "namespace_versioning": True, "fPIC": True}
+    generators = "cmake"
+    exports_sources = ["IlmImf__ImfSystemSpecific.cpp.patch", "CMakeLists.txt.patch"]
```
Comment (uilianries) [inline recipes/openexr/all/conanfile.py]:
```suggestion
    exports_sources = ["*.patch"]
```

**Comment by Croydon** [comment]
https://github.com/conan-community/conan-openexr can be achieved now 😄

**Comment by uilianries** [comment]
Done, it's deprecated!

https://bintray.com/conan-community/conan/openexr%3Aconan
https://github.com/conan-community/conan-openexr

**Pull request merged**